registry/storage/driver/gdrive/gdrive.go

```diff
@@ -0,0 +1,488 @@
+
+	if r != nil && len(r.Files) > 0 {
+		for _, i := range r.Files {
+			if subPath != "/" && strings.HasPrefix(i.Name, subPath) {
+				driveFileInfo.size = 0
+				driveFileInfo.isDir = true
+				return fileInfo{
+					path:          subPath,
+					driveFileInfo: driveFileInfo,
+				}, nil
+			}
+		}
+	}
+
+	return nil, storagedriver.PathNotFoundError{Path: subPath}
+}
+
+// List returns a list of the objects that are direct descendants of the given
+// path.
+func (d *driver) List(ctx context.Context, subPath string) ([]string, error) {
+	r, _ := d.driveService.Files.List().Q(`"` + d.rootID + `" in parents`).Do()
+
+	keys := []string{}
+	checkMap := make(map[string]bool)
+	if r != nil && len(r.Files) > 0 {
+		for _, i := range r.Files {
+
+			if subPath == "/" {
+				index := strings.IndexByte(i.Name[1:], '/')
+				if _, ok := checkMap[i.Name[:index+1]]; !ok {
+					checkMap[i.Name[:index+1]] = true
+					keys = append(keys, i.Name[:index+1])
+				}
+			} else {
+				if strings.HasPrefix(i.Name, subPath) {
+					index := strings.IndexByte(i.Name[len(subPath)+1:], '/')
+					if index != -1 {
+						index += len(subPath)
+						if _, ok := checkMap[i.Name[:index+1]]; !ok {
+							checkMap[i.Name[:index+1]] = true
+							keys = append(keys, i.Name[:index+1])
+						}
+					} else {
+						keys = append(keys, i.Name)
+					}
+				}
+			}
+		}
+	}
+	if subPath != "/" && len(keys) == 0 {
+		return nil, storagedriver.PathNotFoundError{Path: subPath}
+	}
+	return keys, nil
+}
+
+// Move moves an object stored at sourcePath to destPath, removing the original
+// object.
+func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
+	dstFile := &drive.File{}
+	file := d.getFile(sourcePath)
+	if file != nil {
+		dstFile.Name = destPath
+		_, err := d.driveService.Files.Update(file.Id, dstFile).Do()
+		if err != nil {
+			return err
+		}
+	} else {
+		return storagedriver.PathNotFoundError{Path: sourcePath}
+	}
+	return nil
+}
+
+// Delete recursively deletes all objects stored at "path" and its subpaths.
+func (d *driver) Delete(ctx context.Context, subPath string) error {
+	r, _ := d.driveService.Files.List().
+		Q(`"` + d.rootID + `" in parents`).Do()
+	pathFound := false
+	if r != nil && len(r.Files) > 0 {
+		for _, i := range r.Files {
+			if subPath != "/" && strings.HasPrefix(i.Name, subPath) {
+				pathFound = true
+				err := d.driveService.Files.Delete(i.Id).Do()
+				if err != nil {
+					log.Fatalf("Cannot delete the file:%s %s", i.Name, err)
+				}
+			}
+		}
+	}
+	//time.Sleep(time.Second * 1)
+	if !pathFound {
+		return storagedriver.PathNotFoundError{Path: subPath}
+	}
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
+	return "", storagedriver.ErrUnsupportedMethod{}
```

Comment:
Is this possible with Google Drive or would the underlying file first need to be shared?